frontend/apps/pwa/src/components/FormPicker.tsx

```diff
@@ -1,34 +1,69 @@
-import { FC, useCallback } from 'react';
+import React, { FC, useCallback } from 'react';
 import { FormDefinition, FormType } from 'core-api-client';
+import { FieldErrors } from 'react-hook-form';
+import { ErrorMessage } from '@hookform/error-message';
 
 import { useDatabase, useForms } from '../app/hooks';
+import { ValidationForm } from '../reducers/Former/types';
+import { registeredValidation } from '../features/former/validation';
 
 export type FormPickerProps = {
   forms: FormDefinition[];
   formId: string | undefined;
   disabled?: boolean;
   setFormId: (formId: string) => void;
+  register: any;
+  errors?: FieldErrors<ValidationForm>;
 };
 
-export const FormPicker: FC<FormPickerProps> = (props) => {
-  const { forms, formId, setFormId, disabled } = props;
+export const FormPicker: FC<FormPickerProps> = ({
+  forms,
+  formId,
+  setFormId,
+  disabled,
+  register,
+  errors,
+}) => {
   const hasForms = forms.length > 0;
+
+  const registerSelectedFieldReference = register(
+    'selectedField.fieldType.reference.formId',
+    registeredValidation.selectedField.fieldType.reference.formId,
```

Comment:
What is this argument doing? It looks like an initial value, but it doesn't seem to match up with what I read in the docs.

Reply:
this is passing the validation rules for this field to the register function

Reply:
ah yes, I didn't get down to the `registeredValidation` file yet

Reply:
not a good one. will change that.